### PR TITLE
fixes size of the download spinner

### LINKED ISSUE #24802: Download as ZIP spinner out of place
### Steps

1. Upload several files in the web UI
1. Select files
1. Click on "Download" in the toolbar above the column headers

### Expected result
Spinner looks alright

### Actual rresult
Spinner looks out of place / too big

@skjnldsv can you have a look ? Are these different spinner sizes for the new one ?

@owncloud/designers

### PR DESCRIPTION
**Pull request by vincchan**
fixes #24802

### CODE REVIEW

File: core/css/icons.css [hunk 1]
```diff
@@ -66,6 +66,10 @@
 	margin: -9px 0 0 -9px;
 }
 
+.icon-loading-download:after{
```
Comment (skjnldsv) [inline core/css/icons.css]: Unnecessary, you don't need to add css, just use the oc requirements for the owncloud icons (see second message) :)

File: apps/files/js/fileactions.js [hunk 2]
```diff
@@ -665,11 +665,11 @@
 	OCA.Files.FileActions.updateFileActionSpinner = function($downloadButtonElement, showIt) {
 		var $icon = $downloadButtonElement.find('.icon');
 		if (showIt) {
-			var $loadingIcon = $('<span class="icon loading"></span>');
+			var $loadingIcon = $('<span class="icon-loading-small icon-loading-download"></span>');
```
Comment (skjnldsv) [inline apps/files/js/fileactions.js]: you need the icon class whenever you use an ucon (see the dev documentation)

var $loadingIcon = $('<span class="icon icon-loading-small"></span>');

File: apps/files/js/fileactions.js [hunk 3]
```diff
@@ -665,11 +665,11 @@
...
 			$icon.after($loadingIcon);
 			$icon.addClass('hidden');
 		} else {
-			$downloadButtonElement.find('.loading').remove();
+			$downloadButtonElement.find('.icon-loading-download').remove();
```
Comment (skjnldsv) [inline apps/files/js/fileactions.js]: `$downloadButtonElement.find('.icon-loading-small').remove();`

**Comment by skjnldsv** [comment]
:-1: See my comments on the code. :)

File: apps/files/js/fileactions.js (hunk 2 above)
Comment (nickvergessen) [inline apps/files/js/fileactions.js]: :+1:

**Comment by vincchan** [comment]
@skjnldsv Ah I see! :) fixed the code, could you have another look. Thanks!

**Comment by skjnldsv** [comment]
Works perfectly! :+1: :)

**Comment by vincchan** [comment]
@skjnldsv rebased!

**Comment by nickvergessen** [comment]
Works :+1:

**Comment by nickvergessen** [comment]
Javascript tests fail, you can run them with `./autotest-js.sh`

```
PhantomJS 1.9.8 (Linux) OCA.Files.FileActions tests download spinner replaces download icon with spinner FAILED
	Expected 0 to equal 1.
	Error: Expected 0 to equal 1.
	    at /home/nickv/ownCloud/master/core/apps/files/tests/js/fileactionsSpec.js:669
```

```js
		it('replaces download icon with spinner', function() {
			FileActions.updateFileActionSpinner($el, true);
			expect($el.find('.icon.loading').length).toEqual(1);
			expect($el.find('.icon.icon-download').hasClass('hidden')).toEqual(true);
		});
		it('replaces spinner back with download icon with spinner', function() {
			FileActions.updateFileActionSpinner($el, true);
			FileActions.updateFileActionSpinner($el, false);
			expect($el.find('.icon.loading').length).toEqual(0);
			expect($el.find('.icon.icon-download').hasClass('hidden')).toEqual(false);
		});
```

should be easy to adjust

**Comment by nickvergessen** [comment]
Rebased and pushed a fix for the tests

**Comment by PVince81** [comment]
Tired of waiting for Jenkins. Ran the JS Unit tests locally, they pass :+1:

**Pull request merged**